packages/app/src/specs/generators/ExpandableFileChooser.vue

```diff
@@ -0,0 +1,166 @@
+        <ReactComponentList
+          :file="file"
+          @selectItem="selectItem"
+        />
+      </template>
+      <template #no-results>
+        <NoResults
+          empty-search
+          :search-term="noResults.search"
+          :message="noResults.message"
+          @clear="noResults.clear"
+        />
+      </template>
+    </ExpandableFileList>
+  </CreateSpecModalBody>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue'
+import { debouncedWatch, useDebounce, useElementSize } from '@vueuse/core'
+import { useI18n } from '@cy/i18n'
+
+import NoResults from '@cy/components/NoResults.vue'
+import CreateSpecModalBody from './CreateSpecModalBody.vue'
+import ExpandableFileList from './ExpandableFileList.vue'
+import FileMatch from '../../components/FileMatch.vue'
+import ReactComponentList from './component/ReactComponentList.vue'
+import { gql } from '@urql/core'
+
+const props = withDefaults(defineProps<{
+  files: any[]
+  extensionPattern: string
+  loading?: boolean
+}>(), {
+
+  loading: false,
+})
+
+const { t } = useI18n()
+
+gql`
+fragment FileChooser on FileParts {
+  relative
+  id
+  ...FileListItem
+}
+`
+
+const emits = defineEmits<{
+  (eventName: 'selectItem', { file: File, item: string })
+  (eventName: 'update:extensionPattern', value: string)
+}>()
+
+// eslint-disable-next-line
+const initialExtensionPattern = props.extensionPattern
+const localExtensionPattern = ref(props.extensionPattern)
+const filePathSearch = ref('')
+
+const selectItem = ({ file, item }) => {
+  emits('selectItem', { file, item })
+}
+
+///*------- Styling -------*/
+
+// For the FileList to be searchable without jumping to the top of the
+// FileMatcher's top when focused, we need to use some manual layout.
+// If we concede on position: sticky for the FileMatcher or on making the
+// FileList accessible, we could position the FileList relatively.
+const fileMatchRef = ref(null)
+const { height: fileMatchHeight } = useElementSize(fileMatchRef)
+
+///*------- Debounce -------*/
+
+const debounce = 200
+const debouncedExtensionPattern = useDebounce(localExtensionPattern, debounce)
+
+debouncedWatch(localExtensionPattern, (value) => {
+  emits('update:extensionPattern', value)
+}, { debounce })
+
+///*------- Searching files -------*/
+
+const filteredFiles = computed(() => {
+  return props.files?.filter((file) => {
+    return file.relative.toLowerCase().includes(filePathSearch.value.toLowerCase())
+  })
+})
+
+///*------- Matches Indicator -------*/
+
+const matches = computed(() => {
+  return {
+    total: props.files.length,
+    found: filteredFiles.value.length,
+  }
+})
+
+///*------- No Results Options -------*/
+
+const noResults = computed(() => {
+  return {
+    search: filePathSearch.value || debouncedExtensionPattern.value,
+    message: filePathSearch.value ? t('noResults.defaultMessage') : t('components.fileSearch.noMatchesForExtension'),
+    clear: filePathSearch.value ?
```

Comment:
Not sure you wrote this originally but should this be a single fn with a conditional?

```js
clear: () => {
  if (filePathSearch.value) {
    filePathSearch.value = ''"
  } else {
    localExtensionPattern.value = initialExtensionPattern
  }
}
```